src/Examples/VGG.cs

```diff
@@ -0,0 +1,211 @@
+            IEnumerable<(TorchTensor, TorchTensor)> dataLoader, int epoch, long batchSize, long size
+            )
+        {
+            module.Train();
+
+            int batchId = 1;
+            long total = 0;
+            long correct = 0;
+
+            foreach(var (data, target) in dataLoader)
+            {
+                optimizer.ZeroGrad();
+                using (var prediction = module.Forward(data))
+                using (var output = loss(TorchSharp.NN.Module.LogSoftMax(prediction, 1), target))
+                {
+                    output.Backward();
+                    optimizer.Step();
+
+                    var predicted = prediction.Argmax(1);
+                    total += target.Shape[0];
+                    correct += predicted.Eq(target).Sum().DataItem<long>();
+                    if (batchId % _logInterval == 0)
+                    {
+                        Console.WriteLine($"\rTrain: epoch {epoch} [{batchId * batchSize} / {size}] Loss: {output.DataItem<float>()} Acc: { (float)correct / total }");
+                    }
+
+                    batchId++;
+
+                    data.Dispose();
+                    target.Dispose();
+
+                }
+            }
+
+        }
+
+        /// <summary>
+        ///     
+        /// </summary>
+        private class Model : Module
+        {
+            Sequential sequential;
+            Module classifier;
+            private readonly static Dictionary<string, object[]> cfgs = new Dictionary<string, object[]>();
+
+            /// <summary>
+            /// 
+            /// </summary>
+            /// <param name="VGGName"></param>
+            public Model(string VGGName) : base()
+            {
+                object[] vgg11 = new object[] { 64, 'M', 128, 'M', 256, 256, 'M', 512, 512, 'M', 512, 512, 'M' };
+                object[] vgg13 = new object[] { 64, 64, 'M', 128, 128, 'M', 256, 256, 'M', 512, 512, 'M', 512, 512, 'M' };
+                object[] vgg16 = new object[] { 64, 64, 'M', 128, 128, 'M', 256, 256, 256, 'M', 512, 512, 512, 'M', 512, 512, 512, 'M' };
+                object[] vgg19 = new object[] { 64, 64, 'M', 128, 128, 'M', 256, 256, 256, 256, 'M', 512, 512, 512, 512, 'M', 512, 512, 512, 512, 'M' };
+
+                cfgs.Add("VGG11", vgg11);
+                cfgs.Add("VGG13", vgg13);
+                cfgs.Add("VGG16", vgg16);
+                cfgs.Add("VGG19", vgg19);
+
+                object[] cfg = cfgs.GetValueOrDefault(VGGName);
+                this.sequential = MakeSequential(cfg);
+                classifier = Linear(320, 50);
+
+            }
+
+            /// <summary>
+            /// 
+            /// </summary>
+            /// <param name="input"></param>
+            /// <returns></returns>
+            public override TorchTensor Forward(TorchTensor input)
+            {
+                TorchTensor res = this.sequential.Forward(input);
+                res = res.View(new long[] { -1, 0 });
```

Comment:
I think this is wrong. You need to take the size of the first dimension of res.

Reply:
Yes ,
I think this should work
                res = res.View(new long[] { res.Shape[0], -1 });